File: ex/loop5/loop5.c

int n;
int x;
int y;

int main() 
{
	if(n>=0) {
		x = n;
		y = n;
		while(x > 0)
		{
			x=x-1;
			y=y-1;
		}
		assert(x+y == 0);
	}
}

File: ex/loop5/loop5.m
type number : -10..10;
var n : number;
	x : number;
	y : number;
	res : number;


ruleset i : number do startstate "Init"
begin
    n := i;
	x := n;
	y := n;
	res := 0;
endstartstate;endruleset;

rule "loop"
	n >= 0 & x > 0 ==> begin
	x := x - 1;y := y - 1;endrule;

rule "assert"
	n >= 0 & x <= 0 ==> begin
	res := x + y;endrule;
	
	
	
invariant "if"
  n >= 0 -> res = 0;